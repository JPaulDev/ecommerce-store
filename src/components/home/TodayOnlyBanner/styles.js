import styled, { css } from 'styled-components';

const headingStyles = css`
  font-family: 'Uni Sans', sans-serif;
  font-weight: 600;
`;

export const Container = styled.div`
  background-color: #d72140;
  background-image: url(${({ background }) => background});
  background-position: 96% 60%;
  background-repeat: no-repeat;
  border: 1px solid var(--color-grey-9);
  margin: 15px 0;
  overflow: hidden;
  padding: 20px 220px 27px 20px;
  position: relative;

  @media screen and (min-width: 935px) {
    display: flex;
    align-items: center;
    column-gap: 30px;
    background-image: none;
    background-color: white;
    padding: 25px 30px;
  }
`;

export const RedOval = styled.div`
  background: linear-gradient(to right, #c12134, #da2141);
  border-radius: 50%;
  height: 170px;
  left: -67px;
  position: absolute;
  width: 300px;

  @media screen and (max-width: 934px) {
    display: none;
  }
`;

export const LeftHeading = styled.h3`
  color: white;
  font-size: 1.6rem;
  z-index: 10;
  ${headingStyles}

  @media screen and (min-width: 575px) {
    font-size: 1.95rem;
    margin-right: 30px;
  }
`;

export const RightHeading = styled.h3`
  font-size: clamp(20px, 3.48vw, 25px);
  margin-right: auto;
  ${headingStyles}

  @media screen and (min-width: 700px) {
    font-size: 1.6rem;
  }

  @media screen and (max-width: 934px) {
    margin: 5px 0 15px;
  }
`;
