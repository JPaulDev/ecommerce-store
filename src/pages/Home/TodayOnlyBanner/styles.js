import styled from 'styled-components';

export const Container = styled.div`
  background-color: #d72140;
  background-image: url(${({ background }) => background});
  background-position: 96% 60%;
  background-repeat: no-repeat;
  border: 1px solid var(--color-grey-9);
  margin: 15px 0;
  overflow: hidden;
  padding: 20px 220px 27px 20px;
  position: relative;

  @media screen and (min-width: 935px) {
    display: flex;
    align-items: center;
    column-gap: 30px;
    background-image: none;
    background-color: white;
    padding: 25px 30px;
  }
`;

export const RedOval = styled.div`
  background: linear-gradient(to right, #c12134, #da2141);
  border-radius: 50%;
  height: 170px;
  left: -67px;
  position: absolute;
  width: 300px;

  @media screen and (max-width: 934px) {
    display: none;
  }
`;

export const LeftHeader = styled.h3`
  color: white;
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.8rem;
  font-weight: 600;
  line-height: 1;
  z-index: 10;

  @media screen and (min-width: 575px) {
    font-size: 1.95rem;
    margin-right: 30px;
  }
`;

export const RightHeader = styled.h3`
  font-family: 'Uni Sans', sans-serif;
  font-size: clamp(20px, 3.48vw, 25px);
  font-weight: 600;
  margin-right: auto;

  @media screen and (min-width: 700px) {
    line-height: 1;
    font-size: 1.6rem;
  }

  @media screen and (max-width: 934px) {
    margin: 10px 0 15px;
  }
`;
