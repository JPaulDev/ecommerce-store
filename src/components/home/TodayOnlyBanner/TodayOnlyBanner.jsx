import background from '../../../../public/images/today-only-banner/gaming-pc.webp';
import { LinkChevron, LinkButton, ResponsiveBox } from '../../ui';
import * as Styled from './styles';

export default function TodayOnlyBanner() {
  const linkText = 'View All Offers';

  return (
    <Styled.Container background={background.src}>
      <Styled.RedOval />
      <Styled.Heading>
        TODAY ONLY
        <Styled.InnerText>
          Fantastic offers, updated every weekday
        </Styled.InnerText>
      </Styled.Heading>
      <ResponsiveBox sx={{ display: { 0: 'none', 935: 'block' } }}>
        <LinkChevron href="/" fontSize={13} iconSize={18}>
          {linkText}
        </LinkChevron>
      </ResponsiveBox>
      <ResponsiveBox sx={{ display: { 0: 'block', 935: 'none' } }}>
        <LinkButton href="/">{linkText}</LinkButton>
      </ResponsiveBox>
    </Styled.Container>
  );
}
