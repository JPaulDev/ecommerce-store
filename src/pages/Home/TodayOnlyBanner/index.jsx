import useMediaQuery from '../../../hooks/useMediaQuery';
import backgroundImage from '../../../assets/images/today-only-banner/gaming-pc.webp';
import LinkWithChevron from '../../../components/LinkWithChevron';
import LinkButtonGrey from '../../../components/LinkButtonGrey';
import * as Styled from './styles';

export default function TodayOnlyBanner() {
  const isMatch = useMediaQuery('(min-width: 935px)');
  const linkText = 'View All Offers';

  return (
    <Styled.Container background={backgroundImage}>
      <Styled.RedOval />
      <Styled.LeftHeader>TODAY ONLY</Styled.LeftHeader>
      <Styled.RightHeader>
        Fantastic offers, updated every weekday
      </Styled.RightHeader>
      {isMatch ? (
        <LinkWithChevron url="" text={linkText} fontSize="0.8rem" />
      ) : (
        <LinkButtonGrey href="" padding="12px 25px">
          {linkText}
        </LinkButtonGrey>
      )}
    </Styled.Container>
  );
}
