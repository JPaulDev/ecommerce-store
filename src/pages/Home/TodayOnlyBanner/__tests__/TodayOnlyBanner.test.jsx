import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TodayOnlyBanner from '../index';

describe('TodayOnlyBanner component', () => {
  it('renders a today only offers banner', () => {
    const tree = renderer.create(<TodayOnlyBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('TodayOnlyBanner component at different screen widths', () => {
  it('renders a link with a chevron at 935px and above', () => {
    window.resizeTo(935, 768);
    render(<TodayOnlyBanner />);

    expect(screen.getByTestId('link-chevron')).toBeInTheDocument();
    expect(screen.queryByTestId('link-button')).not.toBeInTheDocument();
  });

  it('renders a link button at 934px and below', () => {
    window.resizeTo(934, 768);
    render(<TodayOnlyBanner />);

    expect(screen.queryByTestId('link-chevron')).not.toBeInTheDocument();
    expect(screen.getByTestId('link-button')).toBeInTheDocument();
  });
});
