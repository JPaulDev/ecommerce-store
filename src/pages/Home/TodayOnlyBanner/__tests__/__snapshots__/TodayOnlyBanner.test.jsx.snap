// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodayOnlyBanner component renders a today only offers banner 1`] = `
<div
  className="sc-hKMtZM iuxvJc"
>
  <div
    className="sc-eCYdqJ fxIzfa"
  />
  <h3
    className="sc-jSMfEi gChssM"
  >
    TODAY ONLY
  </h3>
  <h3
    className="sc-gKXOVf bMFknS"
  >
    Fantastic offers, updated every weekday
  </h3>
  <a
    className="sc-gsnTZi iWfFcv"
    data-testid="link-chevron"
    href=""
  >
    View All Offers
    <i
      className="sc-bczRLJ klUGFp material-icons"
    >
      chevron_right
    </i>
  </a>
</div>
`;
